### PR TITLE
add the fixer NoSpaceBeforeComma

### PR DESCRIPTION
**Pull request by clairecoloma**
Hello,

This PR manage spaces before and after a comma. 
Like you can see in tests `Superman    ,  you're my hero` become `Superman, you're my hero`

### CODE REVIEW

**Comment by ternel** [comment]
Yup, plz fix this @clairecoloma :)

File: src/JoliTypo/Fixer/NoSpaceBeforeComma.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * NO_SPACE before ,
+ * As recommended by "Abrégé du code typographique à l'usage de la presse", ISBN: 978-2351130667
+ *
+ * @package JoliTypo\Fixer
+ */
+class NoSpaceBeforeComma implements FixerInterface
+{
+    public function fix($content, StateBag $state_bag = null)
+    {
+        $content = preg_replace('@(\w+) *(,) *'.Fixer::NO_BREAK_SPACE.'*@mu', '$1$2'.Fixer::NO_BREAK_SPACE, $content);
```
Comment (paulmallet) [inline src/JoliTypo/Fixer/NoSpaceBeforeComma.php]: \s pour gérer tous les types de whitespace au lieu de l'espace seul non ?

**Comment by mdarse** [comment]
It would be good to have a test for this ;)

**Comment by mdarse** [comment]
My eyes are broken, now I see the light, I see those tests !

**Comment by damienalexandre** [comment]
Thanks @clairecoloma for this well done PR,
but I'm dubious about the use of a NO_BREAK_SPACE after the comma. There is no sign of it in the French typographic code I use, do you have any source mentionning this usage?

Otherwise, we could just set back a normal space (or the space captured in the regexp).

**Comment by clairecoloma** [comment]
Yeah you're right, we can use a normal space instead of a NO_BREAK_SPACE.

**Comment by ternel** [comment]
IMO, we have to use a normal space, and not a no break space.

**Comment by mdarse** [comment]
Any reason to not keep the original character after the coma?

Mathieu

Le 10 févr. 2015 à 10:29, Benjamin Clay <notifications@github.com> a écrit :

IMO, we have to use a normal space, and not a no break space.

—
Reply to this email directly or view it on GitHub
<https://github.com/jolicode/JoliTypo/pull/13#issuecomment-73668950>.

**Comment by ternel** [comment]
I don't see any reason to put a no break space here, I think that it should really be a simple space. But if there's no specs, yeah, just keep the original character.

File: README.md [hunk 2]
```diff
@@ -107,6 +107,10 @@ FrenchNoBreakSpace
 Replace some classic spaces by non breaking spaces following the French typographic code.
 No break space are placed before `:`, thin no break space before `;`, `!` and `?`.
 
+NoSpaceBeforeComma
+------------------
+Remove space before a `,` and add a non breaking spaces after.
```
Comment (pyrech) [inline README.md]: The description need to be updated too ;)

**Comment by damienalexandre** [comment]
Thanks @clairecoloma I just merged the feature!

**Pull request merged**